symLogging.py

```diff
@@ -1,29 +1,54 @@
 import sys
 import threading
 import time
+import logging
+import os.path
+from logging.handlers import RotatingFileHandler
 
 gEnableTracing = False
+logger = None
 
 # Whether to pause and launch debugger at CheckDebug breakpoints
 gDebug = False
 
-def SetTracingEnabled(isEnabled):
-  global gEnableTracing
+
+def SetTracingEnabled(isEnabled, logOptions):
+  global gEnableTracing, logger
   gEnableTracing = isEnabled
 
+  filepath = logOptions.get('logPath', '/var/log/snappy')
+  rotatingHandler = RotatingFileHandler(
+              filename=os.path.join(filepath, 'snappy.log'),
+              mode='a',
+              maxBytes=int(logOptions.get("maxFileSize", 1024*1024)),
+              backupCount=int(logOptions.get("maxFiles", 10)))
+
+  streamHandler = logging.StreamHandler()
+
+  fmt = logging.Formatter('%(asctime)s:%(levelname)s:%(message)s')
+  rotatingHandler.setFormatter(fmt)
+  streamHandler.setFormatter(fmt)
+
+  logger = logging.getLogger("Snappy")
+  logger.addHandler(rotatingHandler)
+  logger.addHandler(streamHandler)
+  logger.setLevel(logging.INFO)
+
 def LogTrace(string):
   global gEnableTracing
   if gEnableTracing:
     threadName = threading.currentThread().getName().ljust(12)
-    print >> sys.stdout, time.asctime() + " " + threadName + " TRACE " + string
+    logger.info("%s %s", threadName, string)
 
 def LogError(string):
-  threadName = threading.currentThread().getName().ljust(12)
-  print >> sys.stderr, time.asctime() + " " + threadName + " ERROR " + string
+  if logger.isEnabledFor(logging.ERROR):
+    threadName = threading.currentThread().getName().ljust(12)
+    logger.error("%s %s", threadName, string)
 
 def LogMessage(string):
-  threadName = threading.currentThread().getName().ljust(12)
-  print >> sys.stdout, time.asctime() + " " + threadName + "       " + string
+  if logger.isEnabledFor(logging.INFO):
+    threadName = threading.currentThread().getName().ljust(12)
+    logger.info("%s       %s", threadName, string)
```

Comment:
Getting the thread name shouldn't be a lot of computation, but it's your call
Btw, in this patch it's not possible to specifying the logging level so logger.isEnabledFor(logging.INFO) will always be True. We should make it possible to specify the logging level via config file or the debug console